src/ledger.js

```diff
@@ -0,0 +1,131 @@
+// LedgerSystem class
+const BigNumber = require('bignumber.js');
+const Data = require("./database");
+const FIFOQueue = require("./queue");
+const sleep = require('./utils').sleep;
+const groupBy = require('./utils').groupBy;
+const OperationTypes = {
```

Comment:
Probably we dont need this enum because there can't be more types of operations by definition.  In fact there should be no concept of 'Type Of Operation'